fix(admin-ui) Custom batch job activity card

pepijn-vanvlaanderen opened the pull request:
Fixes crash on the batch job activity card when using a custom batch job.

Currently the component uses a hard coded list to determine the operation string 'Import' / 'Export' based on the Medusa core batch job types. However Medusa supports creating custom batch jobs, and then this component crashes.

fPolic reviewed (approved):
LGTM @pepijn-vanvlaanderen!

olivermrbl commented:
@pepijn-vanvlaanderen can I get you to add a changeset with a patch-bump? 

Run `yarn changeset` at root and choose the changed packages. It should suggest admin and admin-ui.

pepijn-vanvlaanderen commented:
@olivermrbl sure thing, I've added the changeset.

olivermrbl commented:
@pepijn-vanvlaanderen – should we revert these changes and add a fallback mechanism that uses the raw batch type for the operation name? As @mortenengel suggested.

It would not cover your specific case but on the other hand it ensures all types can be used. Think this is a good tradeoff.

pepijn-vanvlaanderen commented:
@olivermrbl I do not think you want the batchJob type here as operation name. The current code snippet:
```
const operation = {
    "product-import": BatchJobOperation.Import,
    "price-list-import": BatchJobOperation.Import,
    "product-export": BatchJobOperation.Export,
    "order-export": BatchJobOperation.Export,
 }[batchJob.type]
 ```
 Is just translated to:
 const operation = {
    "product-import": "Import",
    "price-list-import": "Import",
    "product-export": "Export",
    "order-export": "Export",
 }[batchJob.type]
 Therefore my change was just to only give `Import` or `Export` back based on the batchJob type.
 
 Further in the code of the `batchJobDescriptionBuilder` the operation is combined with the entityname, i.e.:
 ```
 let description = ""

  const entityName = batchJob.type.split("-").reverse().pop()

  const twentyfourHoursInMs = 24 * 60 * 60 * 1000

  switch (batchJob.status) {
    case "failed":
      description = `${operation} of ${entityName}s has failed.`
      break
    case "canceled":
      description = `${operation} of ${entityName}s has been canceled.`
      break
      ...
 ```
Then if you would use batchJobType as fallback it would become a bit funny, like `custom-import of customs has failed`.

If we always expect the batchJobType to be of format `${operation}-${entityName}`, I could change my code to accomodate that with something like:
```
let operation = batchJob.type.split("-").pop()
operation = operation.charAt(0).toUpperCase() + operation.slice(1)
```

pepijn-vanvlaanderen commented:
@olivermrbl I have updated this PR with the more generic solution

olivermrbl commented:
> If we always expect the batchJobType to be of format 

Right now, all operations in the core package use the opposite format: `[entity]-[operation]`, so the current changes would lead to unreadable descriptions. E.g. `product-import` would be "Product of product .... "

pepijn-vanvlaanderen commented:
Hi @olivermrbl, you are right, I made an adjustment for this.

olivermrbl commented on packages/admin-ui/ui/src/components/organisms/batch-jobs-activity-list/index.tsx:
```diff
@@ -86,12 +86,8 @@ const BatchJobActivityCard = (props: { batchJob: BatchJob }) => {
-    "product-export": BatchJobOperation.Export,
-    "order-export": BatchJobOperation.Export,
-  }[batchJob.type]
+  let operation = batchJob.type.split("-").pop()
```
@pepijn-vanvlaanderen can you add a small description of the formatting for future reference? Would make it easier for people reading the code to quickly know what to do
